browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -31,33 +31,26 @@ void ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
     return;
   }
 
-  if (ctx->method != "GET") {
-    return;
-  }
-
+  // We either pass a valid redirect_source or initiator_url
   if (ctx->redirect_source.is_valid()) {
     if (ctx->internal_redirect) {
       // Ignore internal redirects since we trigger them.
       return;
     }
 
-    if (net::registry_controlled_domains::SameDomainOrHost(
-            ctx->redirect_source, ctx->request_url,
-            net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES)) {
-      // Same-site redirects are exempted.
-      return;
+    auto filtered_url = ApplyPotentialQueryStringFilter(
+        ctx->redirect_source, ctx->request_url, ctx->method);
+
+    if (filtered_url.has_value()) {
+      ctx->new_url_spec = filtered_url.value().spec();
+    }
+  } else {
+    auto filtered_url = ApplyPotentialQueryStringFilter(
```

Comment:
Why are you duplicating this code here? There is no longer a need for an `else`

Reply:
I think this is more confusing now with some checks outside the method and some checks inside it. Why wouldn't we pass redirect_source and initiator_url into `ApplyPotentialQueryStringFilter`? If `internal_redirect` doesn't apply to ios then you would just always pass false

Reply:
Ok I passed all the params. and doing the check similar to what it was before